docs/contract.rst

```diff
@@ -12,63 +12,90 @@ implementation in order to be considered "compliant".
 Logging
 -------
 
-DEBUBGLOG environment variable points to the debug log file.
-
-1. Access logs to stderr.
-
-2. Error logs to stderr.
-
-3. Log format:
-
-   format = '%(asctime)s.%(msecs)03dZ %(levelname)s %(name)s "%(message)s"'
-   datefmt = "%Y-%m-%d %H:%M:%S"
-
-   which should look like this::
-
-    2016-07-14 01:02:03.456Z INFO app "hello"
-
-4. Generates a request ID if one isn't in the `X-Request-ID` header in the
+The ``DEBUGLOG`` environment variable tells Talisker where to write the debug log file.
+
+1. Log format::
+
+    FORMAT = '%(asctime)s.%(msecs)03dZ %(levelname)s %(name)s "%(message)s"'
+    DATEFMT = "%Y-%m-%d %H:%M:%S"
+
+   Where ``message`` is the formatted log message, with any extra parameters
+   appended as ``key=value`` pairs. The resulting log entry should look like this::
+
+    2016-07-14 01:02:03.456Z INFO app "hello key=value key2=value2"
```

Comment:
This should be 
2016-07-14 01:02:03.456Z INFO app "hello" key=value key2=value2